### PR TITLE
New oma-google-calendar component

### PR DESCRIPTION
**Pull request by andershansander**
New **oma-google-calendar** component that is used and styled like this:

```
<style>
  oma-google-calendar::part(calendar) {
    display: grid;
    column-gap: 12px;
    row-gap: 12px;
    grid-template-columns: repeat(5, 1fr);
  }
 
  oma-google-calendar::part(event) {
    border: 1px solid royalblue;
    height: 100px;
    padding: 12px;
  }
</style>
<h1>Google Calendar component</h1>
<oma-google-calendar
  api-key="RESTRICTED_BUT_PUBLIC_API_KEY"
  calendar-id="sv.swedish#holiday@group.v.calendar.google.com"
  time-min="2022-05-18T13:40:54.144Z"
  time-max="2022-06-17T13:40:54.144Z"
></oma-google-calendar>
```

### CODE REVIEW

File: components/google-calendar/src/helpers.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  <p part="time">${extractHourAndMinutes(date)}</p>
+`
+export const renderEvent = (event, showDate = true) =>
+  html`
+    <div part="event">
+      ${showDate ? datePart(event.start.date || event.start.dateTime) : null}
+      ${event.start.dateTime ? timePart(event.start.dateTime) : null}
+      <p part="summary">${event.summary}</p>
+      <p part="description">
+        ${html`
```
Comment (ehannes) [inline components/google-calendar/src/helpers.js:37]: Currently, we get descriptions like this:
```
Helgdag<br>Om du vill dölja helgdagar går du till Google Kalender-inställningarna &gt; Helgdagar i Sverige
```
So we need to read the descriptions as HTML. I guess that's what we are trying to do with this `${html` call here but unfortunately it's not working.

File: components/google-calendar/src/helpers.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+`
+
+const timePart = (date) => html`
+  <p part="time">${extractHourAndMinutes(date)}</p>
+`
+export const renderEvent = (event, showDate = true) =>
+  html`
+    <div part="event">
+      ${showDate ? datePart(event.start.date || event.start.dateTime) : null}
+      ${event.start.dateTime ? timePart(event.start.dateTime) : null}
```
Comment (ehannes) [inline components/google-calendar/src/helpers.js:34]: Wouldn't we need a similar one for end time?

File: components/calendar.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      }
+      oma-google-calendar::part(description) {
+        color: darkslategray;
+      }
+    </style>
+    <h1>Google Calendar component</h1>
+    <oma-google-calendar
+      api-key="RESTRICTED_BUT_PUBLIC_API_KEY"
+      calendar-id="sv.swedish#holiday@group.v.calendar.google.com"
+      group-events-by="day"
```
Comment (ehannes) [inline components/calendar.html:48]: I'm not sure where the problem lies, but if I remove this `group-events-by` attribute, the first event doesn't display any date, even though it should. I checked the logs and the first event actually has a start date, but there's no `part=date` section in the `event`. so there's probably a bug in the code somewhere.

File: components/google-calendar/src/index.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { component, html, useEffect, useState } from 'haunted'
+
+import { groupByDay, renderEvent, renderCalendarDay } from './helpers'
+
+import { getEvents } from './events'
```
Comment (ehannes) [inline components/google-calendar/src/index.js:5]: Not sure if I like the module name `events` here. Not all concerning events are placed in that module. You have a `renderEvent` in the `helpers` module. Maybe `api` would be a more suitable name, that's what you are incapsulating in that module, isn't it? The API calls. Or maybe `googleCalendar`?

**Review by ehannes — CHANGES_REQUESTED** [review]
Great job! Looking very good in my opinion. Looking forward to start using it and see what it's like working with :smile:

File: components/google-calendar/src/index.js (hunk 4 above)
Comment (andershansander) [reply components/google-calendar/src/index.js:5]: I agree, api or googleCalendar is better. Maybe google-calendar-api.js.

File: components/calendar.html (hunk 3 above)
Comment (andershansander) [reply components/calendar.html:48]: Oh ok, I guess I caused it when I did some refactoring and probably missed that the date disappeared from some events. Maybe I should look into unit testing for web components.

File: components/google-calendar/src/helpers.js (hunk 2 above)
Comment (andershansander) [reply components/google-calendar/src/helpers.js:34]: I forgot about that. =) Will add it.

File: components/google-calendar/src/helpers.js (hunk 1 above)
Comment (andershansander) [reply components/google-calendar/src/helpers.js:37]: Hmm, ok. I noticed it worked for some <u>-tags I saw.  I will look into what happens here.

File: components/calendar.html (hunk 3 above)
Comment (andershansander) [reply components/calendar.html:48]: It was caused by this code: `events.map(renderEvent)` in index.js. I had implemented renderEvent like `const renderEvent = (event, showDate = true)`

So the .map call in index.js passed in the index of each event as the second parameter. So showDate ended up being 0, 1, 2, 3, 4 for the events..

File: components/google-calendar/src/helpers.js (hunk 1 above)
Comment (andershansander) [reply components/google-calendar/src/helpers.js:39]: I hate Tagged Template. 😄 When I change it to `html([event.description || ''])` it works. No idea why.. Maybe it is haunted's implementation of html that is to blame and not Tagged Templates though.

Comment (andershansander) [reply components/google-calendar/src/helpers.js]: It works as expected. It was just my understanding of tagged templates. :)

**Review by ehannes — APPROVED** [review]

**Pull request merged**